woocommerce/google-listings-and-ads
# PMax Assets: Assets form for managing asset group

### Changes proposed in this Pull Request:

This PR implements the "Assets form" part of 📌 [Assets form and related components](https://github.com/woocommerce/google-listings-and-ads/issues/1787#fe-misc) in #1787.

The assets form includes:

- Add `AssetGroupCard` for managing the an asset group
- Validate asset group and show error messages after clicking on the "Launch paid campaign" button
- After clicking on "Launch paid campaign", scroll to the error asset field which is nearest the top of page
- Adjust `CampaignAssetsForm` to validate the campaign values only

Other required changes:

- Add `extendAdapter` prop to `AdaptiveForm` for customizing the adapter of the form context
- Add `className` prop to `AssetField` and also make `numOfIssues` and `subheading` props optional

### Screenshots:

https://user-images.githubusercontent.com/17420811/213707451-e25eea85-b154-4b17-93b3-8dcd682f2458.mp4

### Detailed test instructions:

💡 Please note that this PR doesn't include the API communication of creating/updating the asset group.

1. Go to the campaign creation page.
2. Load an asset group via the final URL.
3. Test the form UI of managing the asset group.
4. Test the form validation and error messages.

### Changelog entry

> Add - Asset group management form for the campaign creation and editing pages.
